packages/frontend-2/components/viewer/anchored-point/Thread.vue

```diff
@@ -545,6 +518,44 @@ onMounted(() => {
     emit('update:expanded', true)
   }
 })
+
+const showBanner = computed(
+  () =>
+    threadResourceStatus.value.isDifferentVersion ||
+    threadResourceStatus.value.isFederatedModel ||
+    hasClickedFullContext.value
+)
+
+const bannerText = computed(() => {
+  if (hasClickedFullContext.value) return 'Viewing full context'
+  if (threadResourceStatus.value.isDifferentVersion)
+    return 'Conversation started in a different version'
+  if (threadResourceStatus.value.isFederatedModel) return 'References multiple models'
+  return ''
+})
+
+const bannerButton = computed(() => {
+  if (hasClickedFullContext.value) {
+    return {
+      text: 'Back',
+      icon: ArrowLeftIcon,
+      action: goBack
+    }
+  }
+  return {
+    text: 'Full context',
+    icon: ArrowUpRightIcon,
+    action: handleContextClick
+  }
+})
+
+const handleContextClick = () => {
+  if (threadResourceStatus.value.isDifferentVersion) {
+    onLoadThreadVersionContext()
```

Comment:
The naming for these 2 onLoadXX functions doesn't fully make sense - it implies that this is something that happens on load, but this is what triggers the load in the first place.

Also: Seems like this whole `handleContextClick` function could be returned from your composable as its only reliant on the internals already available in the composable. Then you wouldn't need to expose those 2 onLoadXX functions at all